File: SwiftMarshal/Marshal Ruby-Java/types/RubySymbol.java
/*
 * Copyright 2012 mfizz.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.mfizz.ruby.types;

/*
 * #%L
 * mfz-ruby-marshal
 * %%
 * Copyright (C) 2012 mfizz
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

/**
 *
 * @author mfizz
 */
public class RubySymbol extends RubyType implements Comparable<RubySymbol> {
    
    // name of symbol w/o the leading ":" char
    private final String name;

    /**
     * Creates a new symbol.  If in Ruby you'd type ":name", you just use the
     * "name" part here and don't include the leading ':' char.
     * @param name 
     */
    public RubySymbol(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public int hashCode() {
        // hash code of a symobl should NOT match a string
        return ":".hashCode() + this.name.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final RubySymbol other = (RubySymbol) obj;
        if ((this.name == null) ? (other.name != null) : !this.name.equals(other.name)) {
            return false;
        }
        return true;
    }

    @Override
    public int compareTo(RubySymbol o) {
        return this.name.compareTo(o.name);
    }

    @Override
    public void inspect(StringBuilder sb, int indent) {
        sb.append(':');
        sb.append(this.name);
    }
}

File: SwiftMarshal/RubySymbol.swift
//
//  RubySymbol.swift
//  SwiftMarshal
//
//  Created by Jonas Zaugg on 25.02.15.
//  Copyright (c) 2015 Jonas Zaugg. All rights reserved.
//

import Foundation

public class RubySymbol : RubyType {//, Comparable {
    
    // name of symbol w/o the leading ":" char
    private final var _name:String
    final var name:String {
        get {
            return _name
        }
    }
    
    /**
    * Creates a new symbol.  If in Ruby you'd type ":name", you just use the
    * "name" part here and don't include the leading ':' char.
    * @param name
    */
    public init(name:String) {
        self._name = name;
    }
    
    public func hashCode() -> Int {
        // hash code of a symobl should NOT match a string
        return ":".hashValue + self.name.hashValue
    }
    
    /*public func equals(obj: AnyObject?) -> Bool {
    if (obj == nil) {
    return false;
    }
    
    if (getClass() != obj.getClass()) {
    return false;
    }
    
    let other:RubySymbol = obj as RubySymbol
    if ((self.name == nil) ? (other.name != nil) : !self.name.equals(other.name)) {
    return false;
    }
    return true;
    }
    
    public func compareTo(o: RubySymbol) -> Int {
    return self.name.compareTo(o.name);
    }*/
    
    public override func inspect(inout sb: NSMutableString, indent: Int) {
        sb.appendString(":");
        sb.appendString(name);
    }
    
    public override var description:String {
        get {
            return "\(name)"
        }
    }
}